add job to test github pages deployment

jluhrsen opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/ovn-kubernetes/ovn-kubernetes/blob/master/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

All changes must adhere to this template to make it easy for reviewers
and preserve rationale/history behind every change
-->

## 📑 Description
<!-- Add a brief description of the pr -->

<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

## Additional Information for reviewers
<!--
What exactly did you change - you may also defer to information
contained in commit messages. At a bare minimum it's worth highlighting
which areas of the code were changed as it's easier to assign reviewers
-->

## ✅ Checks
<!-- Make sure your pr passes the CI checks and do check the following fields as needed - -->
- [ ] My code requires changes to the documentation
- [ ] if so, I have updated the documentation as required
- [ ] My code requires tests
- [ ] if so, I have added and/or updated the tests as required
- [ ] All the tests have passed in the CI <!-- If not leave a comment as to why the CI is red and if you need help understanding what's wrong -->

## How to verify it
<!--
Did you include unit tests? or end-to-end tests?
How can I manually verify that this patch achieves its objective
-->

jluhrsen commented:
/assign @tssurya 

@tssurya , PTAL at this. simple mkdocs command in an action to run on all PRs and would gate the PR if something broke the docs build. also would let you see if a fix for a broken docs build would work in a PR before having to merge it.

I broke this PR with a push to make it use awesome-pages instead of awesome-nav to show a failure which is [here](https://github.com/ovn-kubernetes/ovn-kubernetes/actions/runs/13910955731/job/38924891642?pr=5123). it's not using the right tool which is why the current [Test GitHub Pages Build passed](https://github.com/ovn-kubernetes/ovn-kubernetes/actions/runs/13911285925/job/38925864389?pr=5123). other ovn-ci jobs failed though, but obviously we can ignore those for this PR.

jluhrsen commented:
https://issues.redhat.com/browse/SDN-5758

martinkennelly reviewed (dismissed):
I see the lane is building the site.

tssurya reviewed (approved):
thanks @jluhrsen ! looks good to me

only one point is we have https://github.com/ovn-kubernetes/ovn-kubernetes/blob/master/.github/workflows/static.yml which is what we used to deploy the website after the merge.. -> I think they can't be in same yaml file I suppose? or can they ? cause one of them is on-push and the other is on-pull req

please have a nice section on the CI docs for calling these out would be good but its a nit

jluhrsen commented:
> thanks @jluhrsen ! looks good to me
> 
> only one point is we have https://github.com/ovn-kubernetes/ovn-kubernetes/blob/master/.github/workflows/static.yml which is what we used to deploy the website after the merge.. -> I think they can't be in same yaml file I suppose? or can they ? cause one of them is on-push and the other is on-pull req
> 
> please have a nice section on the CI docs for calling these out would be good but its a nit

yeah, we can do it all in the single static.yml file. I changed the PR to do that and added a small blurb about this in the ci docs

tssurya commented:
> > thanks @jluhrsen ! looks good to me
> > 
> > only one point is we have https://github.com/ovn-kubernetes/ovn-kubernetes/blob/master/.github/workflows/static.yml which is what we used to deploy the website after the merge.. -> I think they can't be in same yaml file I suppose? or can they ? cause one of them is on-push and the other is on-pull req
> > 
> > please have a nice section on the CI docs for calling these out would be good but its a nit
> 
> yeah, we can do it all in the single static.yml file. I changed the PR to do that and added a small blurb about this in the ci docs

Thanks Jamo, will take a look when I'm back from PTO. I want this PR first im before the symlinks PR.

tssurya reviewed (commented):
discussed the file name change and we can merge this

tssurya commented on docs/ci/ci.md:
```diff
@@ -379,3 +380,8 @@ defined in https://github.com/kubernetes-sigs/network-policy-api/tree/master/con
+
+# Documentation Build Check
+
+To catch any potential documentation build breakages which would prevent any docs changes
```
add the info for downloaded md files

jluhrsen commented:
@tssurya , PTAL again.

flavio-fernandes reviewed (approved):
nice. TY @jluhrsen !

tssurya reviewed (approved):
thank you!
